package/network/services/omcproxy/files/omcproxy.init

```diff
@@ -1,62 +1,88 @@
 #!/bin/sh /etc/rc.common
-# Copyright (C) 2010-2014 OpenWrt.org
+# Copyright (C) 2018 OpenWrt.org
 
 START=99
 USE_PROCD=1
 PROG=/usr/sbin/omcproxy
 
-# Uncomment to enable verbosity 
-#OPTIONS="-v" 
+# Uncomment to enable verbosity
+#OPTIONS="-v"
 PROXIES=""
 
-
 omcproxy_add_proxy() {
-	local uplink downlink scope proxy
-	config_get uplink $1 uplink
-	config_get downlink $1 downlink
+	local proxy scope uplink downlinks device up
+
 	config_get scope $1 scope
+	config_get uplink $1 uplink
+	config_get downlinks $1 downlink
+
+	[ -n "$uplink" ] || return
+
+	local status="$(ubus -S call "network.interface.$uplink" status)"
+	[ -n "$status" ] || return
+
+	json_load "$status"
+	json_get_var device l3_device
+	json_get_var up up
+
+	[ -n "$device" -a "$up" = "1" ] || {
+		procd_append_param error "$uplink is not up"
+		return;
+	}
+
+	proxy="$device"
+
+	for downlink in $downlinks; do
+		local status="$(ubus -S call "network.interface.$downlink" status)"
+		[ -n "$status" ] || return
 
-	proxy=""
+		json_load "$status"
+		json_get_var device l3_device
+		json_get_var up up
 
-	network_get_device updev $uplink
-	[ -n "$updev" ] || return 0
+		[ -n "$device" -a "$up" = "1" ] || {
+			procd_append_param error "$downlink is not up"
+			continue;
+		}
 
-	for network in $downlink; do
-		network_get_device downdev $network
-		[ -n "$downdev" ] && proxy="$proxy,$downdev"
+		proxy="$proxy,$device"
 
-		# Disable in-kernel querier while ours is active
-		[ -f /sys/class/net/$downdev/bridge/multicast_querier ] && \
-			echo 0 > /sys/class/net/$downdev/bridge/multicast_querier
+		# Disable in-kernel querier while ours is active, default is 1.
+		[ -f /sys/class/net/$device/bridge/multicast_querier ] && \
+			echo 0 > /sys/class/net/$device/bridge/multicast_querier
 	done
 
 	[ -n "$proxy" ] || return 0
 	[ -n "$scope" ] && proxy="$proxy,scope=$scope"
 
-	PROXIES="$PROXIES $updev$proxy"
-
+	PROXIES="$PROXIES $proxy"
 }
 
-omcproxy_add_trigger() {
-	local uplink downlink
+omcproxy_add_network_triggers() {
+	local uplink downlinks
+
 	config_get uplink $1 uplink
-	config_get downlink $1 downlink
+	config_get downlinks $1 downlink
 
-	for network in $uplink $downlink; do
-		procd_add_interface_trigger "interface.*" $network /etc/init.d/omcproxy restart
+	procd_add_interface_trigger "interface.*" $uplink /etc/init.d/omcproxy reload
```

Comment:
I prefer to keep the procd_add_interface_trigger for the uplink in the for loop as before